File: JCleanArchitectureApp/data/src/main/java/com/woowrale/data/repository/remote/RemoteRepository.java
package com.woowrale.data.repository.remote;

import com.woowrale.domain.model.Contact;

import java.util.List;

public class RemoteRepository {

    private RemoteContactSource remoteContactSource;

    public RemoteRepository(RemoteContactSource remoteContactSource) {
        this.remoteContactSource = remoteContactSource;
    }

    public List<Contact> getContacts(String apiContacts, String source, String query) {
        return remoteContactSource.getContacts(apiContacts, source, query);
    }
}

File: KCleanArchitectureApp/data/src/main/java/com/woowrale/data/repository/remote/RemoteRepository.kt
package com.woowrale.data.repository.remote

import com.woowrale.domain.model.Contact

interface RemoteContactSource {
    fun getContacts(
        apiContacts: String,
        source: String,
        query: String
    ): List<Contact>
}

class RemoteRepository(private val remoteContactSource: RemoteContactSource) {
    fun getContacts(
        apiContacts: String,
        source: String,
        query: String
    ): List<Contact> {
        return remoteContactSource.getContacts(apiContacts, source, query)
    }
}
